repair/repair.cc

```diff
@@ -47,9 +47,27 @@
 
 #include <seastar/core/gate.hh>
 #include <seastar/util/defer.hh>
+#include <seastar/core/metrics_registration.hh>
 
 logging::logger rlogger("repair");
 
+struct node_ops_metrics {
+    seastar::metrics::metric_groups _metrics;
+    uint64_t bootstrap_total_ranges{0};
+    uint64_t bootstrap_finished_ranges{0};
+    node_ops_metrics() {
+        namespace sm = seastar::metrics;
+        _metrics.add_group("node_maintenance_operations", {
+            sm::make_gauge("bootstrap_total_ranges", sm::description("Number of total ranges for bootstrap operation on this shard."),
+                            bootstrap_total_ranges),
+            sm::make_gauge("bootstrap_finished_ranges", sm::description("Number of finished ranges for bootstrap operation on this shard."),
+                            bootstrap_finished_ranges),
+        });
```

Comment:
So, each time we initiate a bootstrap, total_ranges will jump, and finished_ranges will climb until it reaches it? And is a boostrap is aborted both will jump down?

Reply:
> So, each time we initiate a bootstrap, total_ranges will jump, and finished_ranges will climb until it reaches it?

Yes. 

>  And is a boostrap is aborted both will jump down?

No. bootstrap_total_ranges never jump down. If the bootstrap is aborted, the only thing we can do is restart the node. It is better not to jump down so the monitor can know the ranges it is supposed to be. Dropping to zero will consume the monitor even more.

Reply:
I thought we can resume bootstrap. But let's consider regular repair. How should total_ranges behave?

Reply:
We can sum all the finished_ragnes / total_ranges of individual repair.  We have to remove finished_ranges and total_ranges once a individual repair job is finished though.

Reply:
Another option is to use a label with the repair job. That might cause a large number of labels when manager is creating small repair jobs, which isn't good.

Reply:
Exactly.